test/e2erp/realrp_test.go

```diff
@@ -14,4 +23,72 @@ var _ = Describe("Resource provider e2e tests [Real]", func() {
 		var err error
 		Expect(err).NotTo(HaveOccurred())
 	})
+
+	It("should not be possible for customer to mutate an osa scale set", func() {
+
+		ctx := context.Background()
+		ctx = context.WithValue(ctx, api.ContextKeyClientID, os.Getenv("AZURE_CLIENT_ID"))
+		ctx = context.WithValue(ctx, api.ContextKeyClientSecret, os.Getenv("AZURE_CLIENT_SECRET"))
+		ctx = context.WithValue(ctx, api.ContextKeyTenantID, os.Getenv("AZURE_TENANT_ID"))
+
+		logrus.SetLevel(log.SanitizeLogLevel("Debug"))
+		logrus.SetFormatter(&logrus.TextFormatter{FullTimestamp: true})
+		logger := logrus.WithFields(logrus.Fields{"location": c.location, "resourceGroup": c.resourceGroup})
+
+		arg := azureclient.ApplicationResourceGroup(c.resourceGroup, c.resourceGroup, c.location)
+		Expect(arg).NotTo(And(BeNil(), BeEmpty()))
+		logger.Infof("application resource group is %s", arg)
+
+		// update the logger to use the application resource group
+		logger = logrus.WithFields(logrus.Fields{"location": c.location, "resourceGroup": arg})
+
+		mrg, err := azureclient.ManagedResourceGroup(ctx, arg)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(arg).NotTo(And(BeNil(), BeEmpty()))
+		logger.Infof("managed resource group is %s", mrg)
+
+		// update the logger to use the managed resource group
+		logger = logrus.WithFields(logrus.Fields{"location": c.location, "resourceGroup": mrg})
+
+		scaleSets, err := ScaleSets(ctx, logger, c.ssc, mrg)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(scaleSets).NotTo(And(BeNil(), BeEmpty()))
+		Expect(len(scaleSets)).Should(BeNumerically(">=", 3))
+
+		// TODO: get detailed error and match on them since we expect the customer to see errors with Code=ScopeLocked
+		var errs []error
+
+		logger.Info("Updating the scale set instance count")
```

Comment:
Instead of logrus we should be using the ginkgo framework for this. There is a By function that we use to print info.

Reply:
from the docs I only see support for a `GingkoWriter`. Is this what you are referring to?

Reply:
Lets leave the logrus as is and discuss potentially updating to `GingkoWriter` in the next sprint